tools/win/generate_breakpad_symbols.py

```diff
@@ -45,94 +69,70 @@ def mkdir_p(path):
     except OSError as e:
         if e.errno == errno.EEXIST and os.path.isdir(path):
             pass
-        else: raise
+        else:
+            raise
 
 
-def GenerateSymbols(options, binaries):
+async def GenerateSymbols(options, binaries):
     """Dumps the symbols of binary and places them in the given directory."""
-
-    q = queue.Queue()
-    print_lock = threading.Lock()
-
-    def _Worker():
-        while True:
-            binary = q.get()
-
-            if options.verbose:
-                with print_lock:
-                    print("Generating symbols for {0}".format(binary), flush=True)
-                    thread_start = datetime.utcnow()
-
-            dump_syms = os.path.join(options.build_dir, 'dump_syms.exe')
-            syms = GetCommandOutput([dump_syms, binary])
-            module_line = re.match(r"MODULE [^ ]+ [^ ]+ ([0-9A-Fa-f]+) (.*)\r\n", syms)
-            if module_line is None:
-                with print_lock:
-                    print("Failed to get symbols for {0}".format(binary), flush=True)
-                q.task_done()
-                continue
-
-            output_path = os.path.join(options.symbols_dir, module_line.group(2),
-                                       module_line.group(1))
-            mkdir_p(output_path)
-            symbol_file = "%s.sym" % module_line.group(2)[:-4]  # strip .pdb
-            f = open(os.path.join(output_path, symbol_file), 'w') # pylint: disable=consider-using-with
-            f.write(syms)
-            f.close()
-
-            if options.verbose:
-                with print_lock:
-                    thread_end = datetime.utcnow()
-                    elapsed = thread_end - thread_start
-                    print("Completed generating symbols for {}: elapsed time {} seconds".format(
-                        binary, elapsed.total_seconds()), flush=True)
-
-            q.task_done()
-
+    semaphore = asyncio.Semaphore(os.cpu_count())
+    task_list = []
     for binary in binaries:
-        q.put(binary)
-
-    for _ in range(options.jobs):
-        t = threading.Thread(target=_Worker)
-        t.daemon = True
-        t.start()
-
-    q.join()
-
-
-def main():
+        task_list.append(
+            asyncio.ensure_future(ProcessBinary(semaphore, options, binary)))
+    result_list = await asyncio.gather(*task_list)
+    first_warning_printed = False
+    for success, error_message in result_list:
+        if not success:
+            print(error_message, file=sys.stderr)
+            return False
+        if error_message and not first_warning_printed:
+            first_warning_printed = True
```

Comment:
Why only print the first one and not all?

Reply:
Because now it's a lot of extra warning output.

 In fact, the most important thing that we have warnings. In addition we print the first one.

But if we get an error, we will break.